package/MDAnalysis/analysis/hbonds/hbond_analysis.py

```diff
@@ -594,9 +595,13 @@ def __init__(self, universe, selection1='protein', selection2='all', selection1_
         self.distance = distance
         self.distance_type = distance_type  # note: everything except 'heavy' will give the default behavior
         self.angle = angle
-        self.traj_slice = slice(start if isinstance(start, int) else None,  # internal frames are 0 based
-                                stop if isinstance(stop, int) else None,
-                                step)
+        slice_index = []
+        for index in [start, stop, step]:
+            if isinstance(index, numbers.Integral):
+                slice_index.append(int(index))
+            else:
+                slice_index.append(None)
```

Comment:
The worrying thing is that the current code will also crush `4.0` into `None`.  Really we should have some centralised slice index checker that we share across all occasions where we check these things.  Quickly looking at this code, we could probably just skip all of this checking and put the indices straight into the `Reader`.